### DISCUSSION TITLE
Tutorial: Using sprite sheet animations in Axmol Game Engine

### DISCUSSION THREAD

**Discussion by CodeAndWeb** [axmolengine/axmol#1052] [Show and tell]
TexturePacker supported cocos2d-x right from the start - for more than 10 years. It's sad to see that it's not supported officially anymore. However I am very happy that we found Axmol! 

I've just started updating our tutorials and hope that this will give your engine more users. I've also added a dedicated tutorial section for it. We'll add a PhysicsEditor and SpriteIlluminator tutorial soon.

Here's our [Using sprite sheet animations in Axmol Game Engine](https://www.codeandweb.com/texturepacker/tutorials/animations-and-spritesheets-in-axmol-game-engine):

<a href="https://www.codeandweb.com/texturepacker/tutorials/animations-and-spritesheets-in-axmol-game-engine">
<img src="https://user-images.githubusercontent.com/716009/217876417-fb4106c1-5187-41ac-8853-1c0c890c482f.png"/>
</a>

I hope you like it!

**Comment by rh101** [comment]
@CodeAndWeb It's awesome that you're doing this, so thank you!

An additional bit of info that may be useful to anyone reading this is that Axmol also has support for custom sprite sheet formats, so developers can implement their own parser and just register it with the engine.  The default implementation is the PLIST parser, but there is also an implementation for a generic JSON format sprite sheet in the `cpp_tests` project, just to show how it's done.  It's under "SpriteFrameCacheTests" - "Test Generic JSON Atlas".  

It's useful in cases where developers want to use existing sprite sheets in formats other than PLIST, or use their own custom format (such as reading obfuscated/encrypted files) etc..

**Reply by aismann** [reply]
> An additional bit of info that may be useful to anyone reading this is that Axmol also has support for custom sprite sheet formats, so developers can implement their own parser and just register it with the engine. The default implementation is the PLIST parser, but there is also an implementation for a generic JSON format sprite sheet in the `cpp_tests` project, just to show how it's done. It's under "SpriteFrameCacheTests" - "Test Generic JSON Atlas".
> 
> It's useful in cases where developers want to use existing sprite sheets in formats other than PLIST, or use their own custom format (such as reading obfuscated/encrypted files) etc..

Axmol need some more infos of such kind of "additional" features on the documentation too.

**Reply by rh101** [reply]
I recall there being a page that we updated with all that info, along with the section about the custom sprite sheets, but I can't seem to find it any more.  Was it deleted from the repo?

**Reply by halx99** [reply]
> I recall there being a page that we updated with all that info, along with the section about the custom sprite sheets, but I can't seem to find it any more. Was it deleted from the repo?

Here: https://github.com/axmolengine/axmol/blob/dev/APPENDIX.md

**Reply by rh101** [reply]
That's the one! Thank you!

**Reply by aismann** [reply]
> > I recall there being a page that we updated with all that info, along with the section about the custom sprite sheets, but I can't seem to find it any more. Was it deleted from the repo?
> 
> Here: https://github.com/axmolengine/axmol/blob/dev/APPENDIX.md

It should be part of the official documentation like this on cococ2dx too:
https://docs.cocos.com/cocos2d-x/manual/en/sprites/creating.html

**Reply by rh101** [reply]
It may be a good idea to create separate repository for the documentation which can be be added to it over time.  The API for this engine is starting to deviate from Cocos2d-x, so we should have it documented somewhere.

**Comment by CodeAndWeb** [comment]
I don't really see the need for additional sprite sheet formats. The plist format is perfect for the job since it supports all features... or what is missing in your opinion?

- Trimming (keeping the original layout)
- Sprite Aliases (for identical sprites e.g. in animations)
- [Polygon sprites](https://github.com/axmolengine/axmol/commit/ac24612e986ff37e36cfc37bd333b0417bb34c8f) to improve performance and allow tighter packed sprite sheets
- [Anchor Points](https://github.com/axmolengine/axmol/commit/033b9aabf01358c3ebbfa26c71d39d7b94361f6a) 

The only thing currently missing is 9-slice sprites which is something that could be added in the future.

**Reply by rh101** [reply]
Oh no, sorry for the misunderstanding! I wasn't implying that any additional sprite sheet formats are required, but simply meant that support is in the engine for developers to add other formats, if they require that functionality.

**Reply by CodeAndWeb** [reply]
Ah - ok. My fault :-) I guess that would be too much for that tutorial since it's already very too long. I was not even able to describe all sprite sheet features...

**Comment by CodeAndWeb** [comment]
The tutorial is currently missing 2 points that I wanted to write in a separate tutorial. That one is already very long...
Just wanted to know your opinion about them:

1) PVR encryption - When saving PVR files, you can use TexturePacker to encrypt them. This makes it hard(er) for users to rip the sprite sheet images. It's not a strong encryption but should suffice to protect the assets against most thieves. cocos2d-x has native support for decrypting these files.
2) The option to use hardware compression - e.g. in a PVR file you can use ETC2, ASTC and other formats.

I am wondering, which image formats you are using. Do you all use PNG (which would be very portable) or also use hardware compression?

**Reply by rh101** [reply]
For me personally, it's a mix between PNG and pvr3ccz, which are processed via TexturePacker CLI.
